tools/browsers-lists/browsers-list.mjs

```diff
@@ -0,0 +1,30 @@
+import { parseSpecString } from '../browser-matcher/spec-parser.mjs'
+import SpecMatcher from '../browser-matcher/spec-matcher.mjs'
+import browsersSupported from './browsers-supported.json' assert { type: "json" }
+
+export default function browsersList (browsersList, spec = 'chrome@latest') {
+  return spec.split(',')
+    .flatMap(specString => {
+      let { browserName, specOperator, browserVersion } = parseSpecString(specString, null)
+
+      if (browserVersion === 'latest') {
+        const specs = Array.from(browsersList[browserName] || [])
+          .sort((a, b) => Number(b.version) - Number(a.version))
+        return [specs[0]]
+      }
+
+      if (!specOperator) {
+        return Array.from(browsersList[browserName] || [])
+      }
+
+      if (!browserVersion) {
+        browserVersion = 999999
+      }
+
+      const matcher = new SpecMatcher()
+        .include(`${browserName}${specOperator}${browserVersion}`)
+
+      return Array.from(browsersList[browserName] || [])
+        .filter(sauceBrowser => matcher.test(`${browserName}@${sauceBrowser.version}`))
+    })
```

Comment:
I'm not sure if this makes sense inside `flatMap`. This is what `browsersList` should be doing:

1. For every spec in specString, separated by `,`, create a MatcherRule for that spec, add it to the SpecsMatcher.
2. Filter thru the entire browsersSupported list with the final SpecsMatcher, and these are the browser-versions that should be tested in this execution.

Right?

Reply:
No, the browsersList is only responsible for taking in a comma delimited string of browser specs and returning an array of capabilities that can be used in WDIO to connect to SauceLabs or Selenium. It does not have anything to do with browser matching within a test.

Reply:
I am pretty sure this is the exact way it worked for the existing JIL browsersList function but I will check.

Reply:
Yup confirmed. The current browserlist in jil processes each value in the comma delimited list independently. So this is matching that behavior and I think thats what we would want anyways.

Reply:
<img width="602" alt="image" src="https://github.com/newrelic/newrelic-browser-agent/assets/28714891/148dcc18-fb6f-4916-9694-b0fccdff143b">
You're right about existing behavior that's broken: ran JIL with `-b "chrome@<113,chrome@>107"` and it still gave me all 4 because it doesn't consider the rules in tandem. It does look like de-dupe happens through the Set.

Reply:
This should be fixed though, no? Thoughts? @bjfield @metal-messiah

Reply:
I think the whole point of passing multiple browsers to the CLI via a comma delimited list is to run tests against each of those browser specs individually. Additionally, we don't ever pass multiple in any CI and I never do it personally so maybe this is a non-issue.

Reply:
I will fix the dedupe for sure.